locations/jclouds/src/main/java/org/apache/brooklyn/location/jclouds/networking/creator/DefaultAzureArmNetworkCreator.java

```diff
@@ -0,0 +1,147 @@
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.collect.ImmutableMap;
+
+import org.jclouds.azurecompute.arm.AzureComputeApi;
+import org.jclouds.azurecompute.arm.domain.ResourceGroup;
+import org.jclouds.azurecompute.arm.domain.Subnet;
+import org.jclouds.azurecompute.arm.domain.VirtualNetwork;
+import org.jclouds.compute.ComputeService;
+
+import org.apache.brooklyn.config.ConfigKey;
+import org.apache.brooklyn.core.config.ConfigKeys;
+import org.apache.brooklyn.util.collections.MutableMap;
+import org.apache.brooklyn.util.core.config.ConfigBag;
+
+public class DefaultAzureArmNetworkCreator {
+
+    public static final Logger LOG = LoggerFactory.getLogger(DefaultAzureArmNetworkCreator.class);
+
+    private static final String DEFAULT_RESOURCE_GROUP = "brooklyn-default-resource-group";
+    private static final String DEFAULT_NETWORK_NAME = "brooklyn-default-network";
+    private static final String DEFAULT_SUBNET_NAME = "brooklyn-default-subnet";
+
+    private static final String DEFAULT_VNET_ADDRESS_PREFIX = "10.1.0.0/16";
+    private static final String DEFAULT_SUBNET_ADDRESS_PREFIX = "10.1.0.0/24";
+
+    public static ConfigKey<Boolean> AZURE_ARM_DEFAULT_NETWORK_ENABLED = ConfigKeys.newBooleanConfigKey(
+            "azure.arm.default.network.enabled",
+            "When set to true, AMP will create a default network and subnet per Azure region and " +
+                    "deploy applications there (if no network configuration has been set for the application).",
+            true);
+
+    public static void createDefaultNetworkAndAddToTemplateOptionsIfRequired(ComputeService computeService, ConfigBag config) {
+        if (!config.get(AZURE_ARM_DEFAULT_NETWORK_ENABLED)) {
+            LOG.info("azure.arm.default.network.enabled is disabled, not creating default network");
+            return;
+        }
+
+
+        Map<String, Object> templateOptions = config.get(TEMPLATE_OPTIONS);
+
+        //Only create a default network if we haven't specified a network name (in template options or config) or ip options
+        if (config.containsKey(NETWORK_NAME)) {
+            LOG.info("Network config specified when provisioning Azure machine. Not creating default network");
+            return;
+        }
+        if (templateOptions != null && (templateOptions.containsKey(NETWORK_NAME.getName()) || templateOptions.containsKey("ipOptions"))) {
+            LOG.info("Network config specified when provisioning Azure machine. Not creating default network");
+            return;
+        }
+
+        LOG.info("Network config not specified when provisioning Azure machine. Creating default network if doesn't exist");
+
+        AzureComputeApi api = computeService.getContext().unwrapApi(AzureComputeApi.class);
+        String location = config.get(CLOUD_REGION_ID);
```

Comment:
Can only get it from  `template.getLocation()` reliably. If you move the default network creation after `template` is created you'll be able to get it. This will also make the above checks easier - all you need to do is look in the `templateOptions` object.